cmd/cloud-controller-manager/main.go

```diff
@@ -125,20 +82,52 @@ func main() {
 	logs.InitLogs()
 	defer logs.FlushLogs()
 
-	// the flags could be set before execute
-	command.Flags().VisitAll(func(flag *pflag.Flag) {
-		if flag.Name == "cloud-provider" {
-			flag.Value.Set("SampleCloudProviderFlagValue")
-			return
-		}
-	})
 	if err := command.Execute(); err != nil {
 		os.Exit(1)
 	}
 }
 
-func startNodeIpamControllerWrapper(ccmconfig *cloudcontrollerconfig.CompletedConfig, nodeIPAMConfig nodeipamconfig.NodeIPAMControllerConfiguration, cloud cloudprovider.Interface) func(ctx genericcontrollermanager.ControllerContext) (http.Handler, bool, error) {
+func cloudInitializer(config *cloudcontrollerconfig.CompletedConfig) cloudprovider.Interface {
```

Comment:
Does this have to be in main.go?

Reply:
It definitely could be put somewhere else. I was thinking to have all necessary code in main and cloud providers only have to pay attention to main.go. They definitely could rearrange the code. Would you think it would be better to have a separate file for those kind of func?

Reply:
Guessing this would be lost objectionable in the "example". Clearly should not be in our sample.

Reply:
Thanks for review. Added an README.md to further explain.